extensions/cocostudio/timeline/CCSkeletonNode.js

```diff
@@ -0,0 +1,187 @@
+    var type = {
+        p: cc.p,
+        size: cc.size,
+        rect: cc.rect
+    };
+
+    var SkeletonNode = BoneNode.extend(/** @lends ccs.SkeletonNode# */{
+        _subBonesMap: null,
+
+        _squareVertices: null,
+        _squareColors: null,
+        _noMVPVertices: null,
+        _skinGroupMap: null,
+
+        _sortedAllBonesDirty: false,
+        _sortedAllBones: null,
+        _batchedBoneVetices: null,
+        _batchedBoneColors: null,
+        _batchedVeticesCount: null,
+        _batchBoneCommand: null,
+
+        ctor: function(){
+            this._squareVertices = [
+                {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0},
+                {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}
+            ];
+            BoneNode.prototype.ctor.call(this);
+            this._subBonesMap = {};
+
+            this._skinGroupMap = {};
+
+            this._rackLength = this._rackWidth = 20;
+            this._updateVertices();
+            this._rootSkeleton = this;
+        },
+
+        getBoneNode: function(boneName){
+            var item = this._subBonesMap[boneName];
+            if(item)
+                return item;
+            return null;
+        },
+
+        getAllSubBonesMap: function(){
+            return this._subBonesMap;
+        },
+
+        changeSkins: function(boneSkinNameMap){
+            //boneSkinNameMap
+            //suitName
+            if(typeof boneSkinNameMap === "object"){
+                var boneSkin;
+                for(var name in boneSkinNameMap){
+                    boneSkin = boneSkinNameMap[name];
+                    var bone = this.getBoneNode(name);
+                    if(null !== bone)
+                        bone.displaySkin(boneSkin, true);
+                }
+            }else{
+                var suit = this._suitMap[boneSkinNameMap/*suitName*/];
+                if (suit)
+                    this.changeSkins(suit, true);
+            }
+        },
+
+        addSkinGroup: function(groupName, boneSkinNameMap){
+            //todo _skinGroupMap
+            this._skinGroupMap[groupName] = boneSkinNameMap;
+        },
+
+        getBoundingBox: function(){
+            var minx, miny, maxx, maxy = 0;
+            minx = miny = maxx = maxy;
+            var boundingBox = this.getVisibleSkinsRect();
+            var first = true;
+            if(boundingBox.x !== 0 || boundingBox.y !== 0 || boundingBox.width !== 0 || boundingBox.height !== 0){
+                minx = cc.rectGetMinX(boundingBox);
+                miny = cc.rectGetMinY(boundingBox);
+                maxx = cc.rectGetMaxX(boundingBox);
+                maxy = cc.rectGetMaxY(boundingBox);
+                first = false;
+            }
+            var allBones = this.getAllSubBones();
+            for(var bone, i=0; i<allBones.length; i++){
+                bone = allBones[i];
+                var r = cc.rectApplyAffineTransform(bone.getVisibleSkinsRect(), bone.getNodeToParentTransform());
```

Comment:
wrong transform!  
`var r = cc.rectApplyAffineTransform(bone.getVisibleSkinsRect(), bone.getNodeToParentTransform(this));`  or
`var r = cc.rectApplyAffineTransform(bone.getVisibleSkinsRect(), bone.getNodeToParentTransform(bone->getRootSkeleton()));`